### PR TITLE
Pin versions of llvm tools

### LINKED ISSUE #67: LLVM install bring in mismatched libraries and headers
Installing HOOMD in a fresh mambaforge environment pulls in:
```
  + clang                13.0.1  hce30654_0            conda-forge/osx-arm64     128kB
  + clang-13             13.0.1  default_h2cfa9b4_0    conda-forge/osx-arm64       1MB
  + clangxx              13.0.1  default_hf584372_0    conda-forge/osx-arm64     128kB
  + hoomd                 3.3.0  cpu_py39hb778b30_0    conda-forge/osx-arm64       7MB
  + libblas               3.9.0  15_osxarm64_openblas  conda-forge/osx-arm64      13kB
  + libcblas              3.9.0  15_osxarm64_openblas  conda-forge/osx-arm64      13kB
  + libclang-cpp         14.0.6  default_h81a5282_0    conda-forge/osx-arm64     130kB
  + libclang-cpp13       13.0.1  default_h2cfa9b4_0    conda-forge/osx-arm64      13MB
  + libclang-cpp14       14.0.6  default_h81a5282_0    conda-forge/osx-arm64      14MB
  + libgfortran      5.0.0.dev0  11_0_1_hf114ba7_23    conda-forge/osx-arm64      20kB
  + libgfortran5    11.0.1.dev0  hf114ba7_23           conda-forge/osx-arm64     923kB
  + liblapack             3.9.0  15_osxarm64_openblas  conda-forge/osx-arm64      13kB
  + libllvm14            14.0.6  h37c5ba8_0            conda-forge/osx-arm64      26MB
  + libopenblas          0.3.20  openmp_h130de29_1     conda-forge/osx-arm64       4MB
  + llvm-openmp          14.0.4  hd125106_0            conda-forge/osx-arm64     304kB
  + numpy                1.23.1  py39h7df2422_0        conda-forge/osx-arm64       6MB
  + tbb                2021.5.0  h3e96240_1            conda-forge/osx-arm64     129kB
```

`libclang-cpp14=14.0.6` apparently causes HOOMD to search for includes in `mambaforge/lib/clang/14.0.6` which doesn't exist in this installation. 
* `llvm-openmp=14.0.4` created `mambaforge/lib/clang/14.0.4/include/omp.h`
* `clang=13.0.1` created `mambaforge/lib/clang/13.0.1/include` with the needed files.

Proposed fix:

Pin `llvmdev`, `clangdev`, `libclang-cpp`, `libcxx`, `clang`, and `clangxx` all to the same version (14.0.6 is the latest at the moment). @tcmoore3 please make this change in a new pull request from your fork.

### PR DESCRIPTION
**Pull request by tcmoore3**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] ~Reset the build number to `0` (if the version changed )~
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Fixes #67

### CODE REVIEW

**Comment by tcmoore3** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -66,9 +66,9 @@ requirements:
     - {{ pin_compatible('libclang-cpp') }}
```
Comment (joaander) [inline recipe/meta.yaml:66]:
```suggestion
    - libclang-cpp 14.0.6
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -54,8 +54,8 @@ requirements:
     - tbb
     - tbb-devel
     - pybind11-abi
-    - llvmdev
-    - clangdev
+    - llvmdev 14.0.6
```
Comment (joaander) [inline recipe/meta.yaml:57]:
```suggestion
    # Pin all LLVM and clang packages to the latest version with each build.
    # This ensures that HOOMD can find the corresponding header files which
    # conda installs to version specific directories.
    - llvmdev 14.0.6
```

**Review by joaander — APPROVED** [review]
Thanks! I made a few suggestions which I will apply after this review.

Will test the package after merging.

**Pull request merged**